tests/integration/run_test.go

```diff
@@ -13,22 +13,63 @@ func runTests(t *testing.T, when spec.G, it spec.S) {
 
 	var service testutil.TestService
 
-	it.Before(func() {
-		service.Create(t, "nginx")
-	})
-
 	it.After(func() {
 		service.Remove()
 	})
 
 	when("rio run is called", func() {
-		it("should create a service with default specifications", func() {
-			runningPods := service.GetRunningPods()
+
+		it.Before(func() {
+			service.Create(t, "nginx", "-p", "80/http")
+		})
+
+		it("should create a service with default specifications that can scale up", func() {
 			assert.Equal(t, 1, service.GetAvailableReplicas(), "should have one available replica")
 			assert.Equal(t, 100, service.GetCurrentWeight())
 			assert.Equal(t, "nginx", service.GetImage())
-			assert.Contains(t, runningPods, service.App.Name)
-			assert.Contains(t, runningPods, "2/2")
+			assert.Contains(t, service.Logs(), "linkerd-init")
+			assert.Contains(t, service.Exec("env"), "KUBERNETES_SERVICE_PORT")
+			runningPods := service.GetRunningPods()
+			for _, pod := range runningPods {
+				assert.Contains(t, pod, service.App.Name)
+				assert.Contains(t, pod, "2/2")
+			}
+
+			service.GenerateLoad("80s", 120)
+			assert.Greater(t, service.GetAvailableReplicas(), 1, "should have more than 1 available replica")
```

Comment:
Is there a risk of this not happening immediately ?

Reply:
Yes... It happens quickly usually when tests aren't run in parallel, but having more tests == more running in parallel == more pressure on CPU == greater timetoscale :(

Reply:
OK if it gets flakey just add a wait method.